Android build fails with missing header

I am currently in the process of rebasing #6796 following the recent game UI refactors (thanks nerzhul) but I am now getting the following build error
```
In file included from jni/src/script/scripting_server.h:21:
jni/src/script/cpp_api/s_base.h:38:10: fatal error: 'cmake_config.h' file not
      found
#include "cmake_config.h"
```
AFAIK, no such file does exist, only `cmake_config.h.in` & `cmake_config_githash.h.in` since I do not fully understand the reason for these extensions I am not qualified to fix this.

This was introduced quite recently by 30821ad8deb42728fbba9de60dda31033f70aaac though surprisingly this does not seem to affect the regular build.

--- a/src/script/cpp_api/s_base.h
+++ b/src/script/cpp_api/s_base.h
@@ -35,7 +35,7 @@ extern "C" {
 #include "common/c_types.h"
 #include "common/c_internal.h"
 #include "debug.h"
-#include "cmake_config.h"
+#include "config.h"
 
 #define SCRIPTAPI_LOCK_DEBUG
 #define SCRIPTAPI_DEBUG

